editor.planx.uk/src/pages/FlowEditor/lib/store/preview.ts

```diff
@@ -395,189 +396,31 @@ export const previewStore: StateCreator<
         .filter((id) => {
-          if (visited.has(id)) return false;
-
-          visited.add(id);
-
+          // Filter out nodes we've already visited (aka have a breadcrumb for) (eg clones)
           const node = flow[id];
-
-          return (
-            node &&
-            !breadcrumbs[id] &&
-            ((node.edges || []).length > 0 ||
-              (node.type && !SUPPORTED_DECISION_TYPES.includes(node.type)))
-          );
+          return node && !breadcrumbs[id];
         })
         .forEach((id) => {
           const node = flow[id];
 
-          const passport = computePassport();
-
+          // Recursively get children in internal portals
           if (node.type === TYPES.InternalPortal) {
             return nodeIdsConnectedFrom(id);
           }
 
-          const fn = node.type === TYPES.Filter ? "flag" : node.data?.fn;
-
-          const [globalFlag] = collectedFlags(id, Array.from(visited));
-
-          let passportValues = (() => {
-            try {
-              return fn === "flag" ? globalFlag : passport.data?.[fn]?.sort();
-            } catch (err) {
-              return [];
-            }
-          })();
-
-          if (fn && (fn === "flag" || passportValues !== undefined)) {
-            const responses = node.edges?.map((id) => ({
-              id,
-              ...flow[id],
-            }));
-
-            let responsesThatCanBeAutoAnswered = [] as any[];
-
-            const sortedResponses = responses
-              ? responses
-                  // sort by the most to least number of comma-separated items in data.val
-                  .sort(
-                    (a: any, b: any) =>
-                      String(b.data?.val).split(",").length -
-                      String(a.data?.val).split(",").length,
-                  )
-                  .filter((response) => response.data?.val)
-              : [];
-
-            if (passportValues !== undefined) {
-              if (!Array.isArray(passportValues))
-                passportValues = [passportValues];
-
-              passportValues = (passportValues || []).filter((pv: any) =>
-                sortedResponses.some((r) => pv.startsWith(r.data?.val)),
-              );
-
-              if (passportValues.length > 0) {
-                responsesThatCanBeAutoAnswered = (sortedResponses || []).filter(
-                  (r) => {
-                    const responseValues = String(r.data?.val)
-                      .split(",")
-                      .sort();
-                    return String(responseValues) === String(passportValues);
-                  },
-                );
-
-                if (responsesThatCanBeAutoAnswered.length === 0) {
-                  responsesThatCanBeAutoAnswered = (
-                    sortedResponses || []
-                  ).filter((r) => {
-                    const responseValues = String(r.data?.val)
-                      .split(",")
-                      .sort();
```

Comment:
I couldn't quite make sense of this remnant - anyone know why we would be splitting on the COMMA throughout this function? (Modern content patterns for `fn` & `val` use dot-separated values or underscored in the case of flags.)

I've essentially ditched/simplified this in my refactor & think everything is still working as intended?